libs/storage/map_files_downloader.cpp

```diff
@@ -154,37 +154,40 @@ std::string GetAcceptLanguage()
   return locale.m_language + "-" + locale.m_country;
 }
 
-// static
-MetaConfig MapFilesDownloader::LoadMetaConfig()
+downloader::MetaConfig MapFilesDownloader::LoadMetaConfig() const
 {
   Platform & pl = GetPlatform();
   std::string const metaServerUrl = pl.MetaServerUrl();
-  std::string httpResult;
 
+  std::optional<downloader::MetaConfig> metaConfig;
   if (!metaServerUrl.empty())
   {
     platform::HttpClient request(metaServerUrl);
     request.SetRawHeader("X-OM-DataVersion", std::to_string(m_dataVersion));
     request.SetRawHeader("X-OM-AppVersion", pl.Version());
     request.SetRawHeader("Accept-Language", GetAcceptLanguage());
     request.SetTimeout(10.0);  // timeout in seconds
+
+    std::string httpResult;
     request.RunHttpRequest(httpResult);
+    metaConfig = downloader::ParseMetaConfig(httpResult);
   }
 
-  std::optional<MetaConfig> metaConfig = downloader::ParseMetaConfig(httpResult);
   if (!metaConfig)
   {
     metaConfig = downloader::ParseMetaConfig(pl.DefaultUrlsJSON());
     CHECK(metaConfig, ());
-    LOG(LWARNING, ("Can't get meta configuration from request, using default servers:", metaConfig->m_serversList));
+    LOG(LWARNING, ("Can't get meta configuration from request, using default servers:", metaConfig->servers));
   }
-  CHECK(!metaConfig->m_serversList.empty(), ());
+
   return *metaConfig;
 }
 
-void MapFilesDownloader::GetMetaConfig(MetaConfigCallback const & callback)
+void MapFilesDownloader::GetMetaConfig(MetaConfigCallback && callback)
 {
-  callback(LoadMetaConfig());
+  auto metaConfig = LoadMetaConfig();
+  GetPlatform().RunTask(Platform::Thread::Gui, [callback = std::move(callback), metaConfig = std::move(metaConfig)]()
+  { callback(static_cast<downloader::MetaConfig>(std::move(metaConfig))); });
```

Comment:
Why are you making ```GetPlatform().RunTask(Platform::Thread::Gui``` here if you delegate this logic to the client's callback? (RunMetaConfigAsync)